File: server/libs/storage-redis/src/main/java/com/sharetimer/storage/redis/config/TimerRedisProps.java
package com.sharetimer.storage.redis.config;

import org.springframework.boot.context.properties.ConfigurationProperties;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@ConfigurationProperties(prefix = "timer.redis")
@RequiredArgsConstructor
@Getter
public class TimerRedisProps {
  
  private final Expiration expiration;
  private final PubSub pubSub;

  @Getter
  @RequiredArgsConstructor
  public static class Expiration {
    private final int dbIndex;
    private final String keyPrefix;
  }

  @Getter
  @RequiredArgsConstructor
  public static class PubSub {
    private final int dbIndex;
    private final String targetTimeUpdatedChannel;
    private final String targetTimeUpdatedMessageType;
    private final String timestampAddedChannel;
    private final String timestampAddedMessageType;
  }

}

File: server/libs/storage-redis/src/main/kotlin/com/sharetimer/storage/redis/config/TimerRedisProps.kt
package com.sharetimer.storage.redis.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "timer.redis")
data class TimerRedisProps(
    val expiration: Expiration,
    val pubSub: PubSub,
) {
    data class Expiration(
        val keyPrefix: String,
    )

    data class PubSub(
        val targetTimeUpdatedChannel: String,
        val targetTimeUpdatedMessageType: String,
        val timestampAddedChannel: String,
        val timestampAddedMessageType: String,
    )
}
